modules/core/src/main/scala/org/tesselation/infrastructure/gossip/GossipDaemon.scala

```diff
@@ -53,10 +57,12 @@ object GossipDaemon {
         .fromQueueUnterminated(rumorQueue)
         .evalMap(validateHash)
         .evalMap(validateSignature)
+        .evalMap(validateSession)
         .evalMap(rumorStorage.addRumors)
-        .flatMap(batch => Stream.iterable(batch))
+        .map(_.sortBy(_._2))
+        .flatMap(Stream.iterable)
         .filter { case (_, signedRumor) => signedRumor.value.origin =!= nodeId }
-        .parEvalMapUnordered(cfg.maxConcurrentHandlers)(handleRumor)
+        .parEvalMap(cfg.maxConcurrentHandlers)(handleRumor)
```

Comment:
If we don't group by type we cannot `parEvalMap`, because the ordering of handled messages won't be preserved. Let's revert it to a regular `evalMap` and we will figure it out later on how to parallelize message types.